enginetest/queries/queries.go

```diff
@@ -5142,6 +5142,42 @@ Select * from (
 			{"b"},
 		},
 	},
+	{
+		Query: `SELECT if(1, 123, 456)`,
+		Expected: []sql.Row{
+			{123},
+		},
+	},
+	{
+		Query: `SELECT if(0, 123, 456)`,
+		Expected: []sql.Row{
+			{456},
+		},
+	},
+	{
+		Query: `SELECT if(0, "abc", 456)`,
+		Expected: []sql.Row{
+			{456},
```

Comment:
This should be a string, but the TestEvaluation doesn't go through the schema[0].Convert code path.

Reply:
should we move to brokenQueries?

Reply:
or is it just bad test setup

Reply:
I think it's just bad test setup, when the results are written to the shell we use the writerIters, but not for evaluation.
I can add it in a separate PR, and see how many failures we get